### ISSUE TITLE
Could you explain (checking contacts if they got whatsapp)

### ISSUE THREAD

**Issue by yazeed44** [ymadkour/SecureWhatsapp#1]
?

**Comment by ymadkour** [comment]
It is same like the already known "whatsapp" you can only chat with the users that you have in your phone contact list. The point here was to make sure that the user had his/her privacy.  Therefore we implemented an option that the user has to send request to other users that have the app, but before to send the requests we needed to check which users,that are on the user phone contact list have the app.
I hope i answered your question.
(BTW unfortunately this version of app on this repository is incomplete )

**Comment by yazeed44** [comment]
Thanks for replay

i'm working on whatsapp for pc on java , when the user add a number , i want to check if this number has whatsapp ? 

is there a way to achive that ? i read your code but i didn't get that at all

**Comment by ymadkour** [comment]
yes, here we used a centralised server. So we stored all the whatsapp user on the server. When we launch the application we retrieve all the user that use the whatsapp and are on the phone contact list.
if you are using p2p it could be done too

**Comment by yazeed44** [comment]
Could you explain how to do it in p2p ? 

i want the user to type a number to the program then the program check if this number has whatsapp , that's all of it

I'm really sorry for disturbing you

**Comment by ymadkour** [comment]
you can use tomp2p library (http://tomp2p.net), which will help you with p2p. I think what you need to do is to send a broadcast message with the number, and see if someone will reply if this number exists or not .

**Comment by yazeed44** [comment]
Yeah i have thought of that before , but i don't want to send to every contact the user add them a message , isn't there a way to check in whatsapp servers ?

**Comment by ymadkour** [comment]
In case you want to use a centralised server then yes there is a way, what you can do is to store all the numbers  and maybe the ip of the user on the server and then you can check if they are stored on it or not and fetch their ip to start the p2p connection.
But in case you don't want it to be centralised then the broadcast message i think is the only way since there is no storage except the local one on the pc.
